docs/index.md

```diff
@@ -145,9 +140,57 @@ illustrates how to do this:
     	$ unset DOCKER_TLS_VERIFY
       ```
 
+## Discovery (optional)
+
+Instead of providing a comma-separated list of machines, Swarm can be configured to
+dynamically discover machines.
+
+This allows to add and remove machines from the cluster without having to reconfigure and restart
+the Swarm manager.
+
+For this to work, each machine will run a Swarm agent. The agent registers the referenced
+Docker daemon, monitors it, and updates the discovery backend with the machine's status.
+
+Swarm supports a wide range of discovery backends, including the `Docker Hub`, `consul`, `zookeeper` and `etcd`.
+
+The following example uses the Docker Hub based `token` discovery service:
+
+1. Create a Swarm cluster using the `docker` command.
+
+    ```bash
+    $ docker run --rm swarm create
+    6856663cdefdec325839a4b7e1de38e8 # <- this is your unique <cluster_id>
+    ```
+
+    The `create` command returns a unique cluster ID (`cluster_id`). You'll need
+    this ID when starting the Swarm agent on a machine.
+
+2. Log into **each machine** and register the Swarm agents to the discovery service.
+
+    The machine's IP must be accessible from the Swarm Manager.
+    Use the following command and replace with the proper `machine_ip` and `cluster_id` to start an agent:
+
+        ```bash
+        docker run -d swarm join --addr=<machine_ip:2375> token://<cluster_id>
```

Comment:
Should the port number be fixed `2375` or a generic `:machine_port`? (Thinking out loud here)